src/arrayMethodSort.js

```diff
@@ -4,8 +4,24 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = (a, b) => {
+      if (a === b) {
+        return 0;
+      }
+
+      return String(a) > String(b) ? 1 : -1;
+    },
+  ) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - 1; j++) {
+        if (compareFunction(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
+    }
```

Comment:
The current sorting algorithm is a bubble sort, which has O(n^2) complexity. This is not efficient for larger arrays. If performance is a concern, consider implementing a more efficient sorting algorithm like quicksort or mergesort.